src/server/qgsfeaturefilter.h

```diff
@@ -0,0 +1,63 @@
+/***************************************************************************
+                              qgsfeaturefilter.h
+                              ------------------
+  begin                : 26-10-2017
+  copyright            : (C) 2017 by Patrick Valsecchi
+  email                : patrick dot valsecchi at camptocamp dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSFEATUREFILTER_H
+#define QGSFEATUREFILTER_H
+
+#include "qgsfeaturefilterprovider.h"
+#include "qgis_server.h"
+
+#include <QMap>
+
+class QgsExpression;
+
+/**
+ * \ingroup server
+ * \class QgsAccessControl
```

Comment:
QgsFetureFilter

Reply:
BTW, I'm not a big fan of such a generic class name. Unless this class has a chance to be moved to `core`, I'd (mildly) suggest to rename to QgsServerFeatureFilter or something like that.

Reply:
I've looked around and the other classes in server are far from being all prefixed by QgsServer. And it's a very generic child class of QgsFeatureFilterProvider.